### DISCUSSION TITLE
Forms Shell and MAUI

### DISCUSSION THREAD

**Discussion by paul-charlton** [xamarin/Xamarin.Forms#11282] [General] [answered]
Hi,
I was wondering whether there was going to be something similar to Shell in MAUI?

We're starting a new project and contemplating Shell, however, with MAUI on the horizon, keeping an upgrade path open to us is important.  Obviously the more painless this is the better!

Thanks n advance,
Paul.

**Comment by domagojmedo** [comment, marked as answer]
I was under the impression that everything that you can do in XF, you will also be able to do in MAUI.

**Reply by davidortinau** [reply]
That's correct! 

Shell v2 is also planned for early phase for .NET MAUI. This will include things such as:

* feature parity for Windows and macOS 
* drop-in xplat alternatives [AppBar](https://github.com/dotnet/maui/issues/152) and [TabView](https://github.com/dotnet/maui/issues/151) that are much more flexible and template-able
* non-visual routes and alternate structures in AppShell.xaml
* better support for things like multiple entry points: Login vs Main, Onboarding vs Main etc.

**Comment by davidortinau** [comment]
@paul-charlton smooth migration is a key goal. The known transition steps are listed in the [FAQ here](https://github.com/dotnet/maui/wiki/FAQs#will-my-xamarinforms-projects-automatically-update-to-net-maui).

Please be sure to chime in on [dotnet/maui proposals](https://github.com/dotnet/maui/issues) where some changes are being suggested that would require more manual migration. If the benefit is there, we'd like to hear it. If it's not, we want to hear that too! We'll also be conducting 1:1 interviews, so it's not only about the GitHub votes/reactions/comments.

**Comment by MhAllan** [comment]
Oh please don't do that. and if you do it keep all alternatives ready because the idea of templating UI and mixing UI with non-UI functionality like routing is a disaster. example, I can't use routing because Shell tabs loading on Android is very slow, see the terrible coupling?
